### PR TITLE
Feature/cluster comm logging

### PR DESCRIPTION
**Pull request by ObiWahn**
Add new logging facility for clustercomm. This is enabled when the new `CLUSTERCOMM` logtopic is set to `DEBUG` or `TRACE`.

```
==> starter_bullshit/coordinator8529/arangod.log <==
2019-05-13T07:28:16Z [20145] DEBUG [23456] {clustercomm} id: 0        --> http://localhost:8550/_db/_system/_api/document?collection=s2010035&waitForSync=false&returnNew=false&returnOld=false&isRestore=false&overwrite=false -- url unkown -- payload: '{"_key":"5010015","client":{"httpConnections":1,"connectionTime":{"sum":0,"count":0,"counts":[0,0,0,0]},"totalTime":{"sum":0.07098722457885742,"count":2,"counts":[1,0,1,0,0,0,0]},"requestTime":{"sum":1.621246337890625e-5,"count":2,"counts":[2,0,0,0,0,0,0]},"queueTime":{"sum":0,"count":0,"counts":[0,0,0,0,0,0,0]},"ioTime":{"sum":0.07097101211547852,"count":2,"counts":[1,0,1,0,0,0,0]},"bytesSent":{"sum":451,"count":2,"counts":[2,0,0,0,0,0]},"bytesReceived":{"sum":220,"count":2,"counts":[2,0,0,0,0,0]}},"clusterId":"CRDN-6c5b9314-2af9-4d92-b340-a7e288eaee82","http":{"requestsTotal":6,"requestsAsync":0,"requestsGet":6,"requestsHead":0,"requestsPost":0,"requestsPut":0,"requestsPatch":0,"requestsDelete":0,"requestsOptions":0,"requestsOther":0},"server":{"uptime":32.76852869987488,"physicalMemory":33695162368,"v8Context":{"available":2,"busy":0,"dirty":0,"free":2,"max":16},"threads":{"scheduler-threads":2,"queued":0},"ttl":{"runs":0,"documentsRemoved":0,"limitReached":0}},"system":{"minorPageFaults":56244,"majorPageFaults":0,"userTime":1.18,"systemTime":0.23,"numberOfThreads":26,"residentSize":225665024,"residentSizePercent":0.006697252903411206,"virtualSize":2202214400},"time":1.5577324969558249e+9}'                                                        
2019-05-13T07:28:16Z [20145] DEBUG [23456] {clustercomm} id: 56       <-- tcp://localhost:8550 -- PRMR-9d002651-9911-424c-8dda-a7eaa04739fc:s2010035 -- payload: '{"_id":"_statisticsRaw/5010015","_key":"5010015","_rev":"_Yp-VjiK---"}'
```

### CODE REVIEW

**Comment by ObiWahn** [comment]
http://jenkins01.arangodb.biz:8080/view/PR/job/arangodb-matrix-pr/

**Comment by ObiWahn** [comment]
http://jenkins01.arangodb.biz:8080/view/PR/job/arangodb-matrix-pr/4426/

File: arangod/Cluster/ClusterComm.cpp [hunk 1]
```diff
@@ -38,10 +38,54 @@
...
 
+
+namespace {
+std::stringstream createRequestInfo(NewRequest const& request) {
+  bool trace = Logger::CLUSTERCOMM.level() == LogLevel::TRACE;
+  std::stringstream ss;
+  ss << "id: " << std::setw(8) << std::setiosflags(std::ios::left)
+               << request._ticketId << std::resetiosflags(std::ios::adjustfield)
+     << " --> " << request._destination
+     << " -- " << ( request._request->fullUrl().empty() ? "url unkown" : request._request->fullUrl())
```
Comment (jsteemann) [inline arangod/Cluster/ClusterComm.cpp]:
```suggestion
     << " -- " << ( request._request->fullUrl().empty() ? "url unknown" : request._request->fullUrl())
```

Comment (jsteemann) [reply arangod/Cluster/ClusterComm.cpp]: I think it would make sense to also show the HTTP method used (e.g. GET, POST, PUT) here.

**Comment by ObiWahn** [comment]
http://jenkins01.arangodb.biz:8080/view/PR/job/arangodb-matrix-pr/4430/

**Pull request merged**